homeassistant/components/co2signal/config_flow.py

```diff
@@ -0,0 +1,249 @@
+TYPE_SPECIFY_COUNTRY = "Specify country code"
+
+
+def _get_entry_type(config: dict) -> str:
+    """Get entry type from the configuration."""
+    if CONF_LATITUDE in config:
+        return TYPE_SPECIFY_COORDINATES
+
+    if CONF_COUNTRY_CODE in config:
+        return TYPE_SPECIFY_COUNTRY
+
+    return TYPE_USE_HOME
+
+
+def _validate_info(hass, config: dict) -> dict:
+    """Validate the passed in info."""
+    if CONF_COUNTRY_CODE in config:
+        latitude = None
+        longitude = None
+    else:
+        latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+        longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+
+    try:
+        data = CO2Signal.get_latest(
+            config[CONF_API_KEY],
+            config.get(CONF_COUNTRY_CODE),
+            latitude,
+            longitude,
+            wait=False,
+        )
+
+    except ValueError as err:
+        err_str = str(err)
+
+        if "Invalid authentication credentials" in err_str:
+            raise InvalidAuth from err
+        if "API rate limit exceeded." in err_str:
+            raise APIRatelimitExceeded from err
+
+        _LOGGER.exception("Unexpected exception")
+        raise UnknownError from err
+    except Exception as err:  # pylint: disable=broad-except
+        _LOGGER.exception("Unexpected exception")
+        raise UnknownError from err
+
+    else:
+        if data.get("status") != "ok":
+            _LOGGER.exception("Unexpected response: %s", data)
+            raise UnknownError
+
+    return data
+
+
+class CO2Error(HomeAssistantError):
+    """Base error."""
+
+
+class InvalidAuth(CO2Error):
+    """Raised when invalid authentication credentials are provided."""
+
+
+class APIRatelimitExceeded(CO2Error):
+    """Raised when the API rate limit is exceeded."""
+
+
+class UnknownError(CO2Error):
+    """Raised when an unknown error occurs."""
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Co2signal."""
+
+    VERSION = 1
+    _data: dict | None
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        data = {CONF_API_KEY: import_info[CONF_TOKEN]}
+
+        if CONF_COUNTRY_CODE in import_info:
+            data[CONF_COUNTRY_CODE] = import_info[CONF_COUNTRY_CODE]
+            new_entry_type = TYPE_SPECIFY_COUNTRY
+        elif (
+            CONF_LATITUDE in import_info
+            and import_info[CONF_LATITUDE] != self.hass.config.latitude
+            and import_info[CONF_LONGITUDE] != self.hass.config.longitude
+        ):
+            data[CONF_LATITUDE] = import_info[CONF_LATITUDE]
+            data[CONF_LONGITUDE] = import_info[CONF_LONGITUDE]
+            new_entry_type = TYPE_SPECIFY_COORDINATES
+        else:
+            new_entry_type = TYPE_USE_HOME
+
+        for entry in self._async_current_entries(include_ignore=True):
+            if entry.source == config_entries.SOURCE_IGNORE:
+                continue
```

Comment:
We can exclude ignored entries from the beginning instead by passing `include_ignore=False`.

Reply:
Nice, I can fix 👍